### DISCUSSION TITLE
A revived SEE begs the question.

### DISCUSSION THREAD

**Discussion by Whammo** [jkotlinski/durexforth#479] [Ideas]
Of an editor that traverses code memory on the fly. You can ```see``` and *perhaps* edit and insert code.
 There is no edit buffer other than the screen.
  New code would be appending from ```here```, a load always compiles and a save decompiles from ```marker``` to ```here```.

**Comment by burnsauce** [comment]
I love the crap out of the Jupiter Ace for this, although it goes about it differently.

**Comment by jkotlinski** [comment]
@Whammo Are you describing the interpreter? :-)

Indeed, the new `SEE` is intended to be used interactively, like the Jupiter Ace.
I have not used the Jupiter Ace though, I suspect there is more to it.

**Reply by Whammo** [reply]
It could be the interpreter. It could be as simple as being able to traverse the wordlist with cursor keys.
 I'm not entirely familiar with how the Jupiter Ace does things, but I'm not suggesting using a block edit.

**Reply by jkotlinski** [reply]
The Jupiter Ace editor, more or less, allows typing `SEE WORD` and editing the output, then press RETURN to recompile. It does not have a block editor.

The setup is compelling, it could be used as a nice graphing calculator.
Those who wish to make larger programs would probably find it very frustrating.

**Reply by jkotlinski** [reply]
A video is better than 1000 words... https://www.youtube.com/watch?v=Ld-555GGZTM

**Reply by jkotlinski** [reply]
and a video that gives a peek of the edit/recompile process: https://www.youtube.com/watch?v=NPNeijpC17I

**Reply by Whammo** [reply]
Having a separate input and output window is both primitive and forward looking.

**Comment by jkotlinski** [comment]
Further pondering: one particularly nice thing about the Jupiter Ace is the split-screen.
In the upper part, there is the output of the program.
In the bottom, there is the space for editing text.
The space for editing text grows or shrinks dynamically, depending on what is being displayed.

For durexForth, it could be particularly nice to program high-res graphics this way. In the lower part, a text editor, in the upper part, the image that is being programmed.

**Reply by Whammo** [reply]
Yes, a real split screen could be very cool.
 Getting durexForth off of the CIA timer and onto the raster interrupt is another thing entirely.  😄
